selfdrive/loggerd/loggerd.cc

```diff
@@ -187,20 +188,20 @@ void encoder_thread(const LogCameraInfo &cam_info) {
         if (!sync_encoders(&s, cam_info.type, extra.frame_id)) {
           continue;
         }
-      }
 
-      if (cam_info.trigger_rotate && (cnt >= SEGMENT_LENGTH * MAIN_FPS)) {
-        // trigger rotate and wait logger rotated to new segment
-        ++s.waiting_rotate;
-        std::unique_lock lk(s.rotate_lock);
-        s.rotate_cv.wait(lk, [&] { return s.rotate_segment > cur_seg || do_exit; });
+        const int frames_per_seg = SEGMENT_LENGTH * MAIN_FPS;
+        if (extra.frame_id >= (cur_seg * frames_per_seg) + s.start_frame_id) {
```

Comment:
This seems wrong, it looks like it checks if `extra.frame_id` is higher than the first frame id of the current segment?